chainio/clients/avsregistry/reader.go

```diff
@@ -119,26 +120,26 @@ func BuildAvsRegistryChainReader(
 ) (*AvsRegistryChainReader, error) {
 	contractRegistryCoordinator, err := regcoord.NewContractRegistryCoordinator(registryCoordinatorAddr, ethClient)
 	if err != nil {
-		return nil, err
+		return nil, types.WrapError(errors.New("Failed to create contractRegistryCoordinator"), err)
```

Comment:
nit for @shrimalmadhur  for later 
- maybe this should not live in `types` should be in utils
- also interface should accept string and an error - so user doesn't need to do `errors.New`

not blocking this PR.

Reply:
created https://github.com/Layr-Labs/eigensdk-go/issues/116